extension/user/auth-strategy/facebook.js

```diff
@@ -12,6 +13,40 @@ module.exports = async (context, input) => {
     return {}
   }
 
-  context.log.error('Facebook strategy is not implemented')
-  throw new InvalidCredentialsError()
+  const clientCredentials = context.config.credentials
+  const authUrl = context.config.magentoUrl + '/auth/token'
+  const { parameters: { success, profile } } = input
+  const log = context.log
+  const storages = context.storage
+  const request = context.tracedRequest('magento-user-extension:facebook-login', { log: true })
+
+  if (!success) {
+    throw new InvalidCredentialsError()
+  }
+
+  const th = new TokenHandler(clientCredentials, authUrl, storages, log, request, !context.config.allowSelfSignedCertificate)
+
+  const magentoTokenResponse = await login(th, profile, input.strategy)
+  await new Promise((resolve, reject) => {
```

Comment:
technically `await` already returns a promise. You are not catching the error, so just doing ` await th.deleteGuestTokens()` will qual this statement. I don't think it being a CB should be a problem, right?